Update container for representer

mpizenberg opened the pull request:
It was not compiling on linux for me so I changed a few things. I also took the opportunity to make the docker build multi-step, and extracted the parser from elm-test-rs and published a tool called elm-strip-comments to replace the usage of cloc we had here.

I hope those changes will fix the current CI failures.

mpizenberg commented:
Seems like there is still a CI issue. @ErikSchierboom could you have a look?

ErikSchierboom commented:
@mpizenberg I _think_ this is due to the image not yet being supported on ECR. I've created a PR to add support: https://github.com/exercism/terraform/pull/17

ErikSchierboom commented:
@mpizenberg The above PR was merged and I've re-run the workflow. It now passes!

mpizenberg commented:
Awesome, I'll reset publish on push then

mpizenberg commented:
@ceddlyburge could you check the ps1 script for Windows is still working. If that's the case, we can merge it.

ceddlyburge commented on bin/run-in-docker.ps1:
```diff
@@ -30,4 +30,4 @@ param (
 )
 
 docker build -t elm-representer .
-docker run --network none --read-only --mount type=bind,src=${InputDirectory},dst=/solution --mount type=bind,src=${OutputDirectory},dst=/output elm-representer $Exercise /solution /output
```
I think the --readonly flag should remain for security reasons?

mpizenberg replied:
I've discussed in the slack about the new setup for containers. There is no more read-only constraint as it proved difficult for other tracks. All modifications are discarded after each run though, just as with docker --rm. For us it enabled simpler code in run.sh. That's why I removed the --readonly param in the run-in-docker scripts

mpizenberg replied:
By the way, I also did the same in the elm-test-runner repo

ceddlyburge replied:
Yeah I don't think its important, but I was asked to add it in the initial PR ...

ceddlyburge reviewed (approved):
The ps1 script didn't work, as the parameters names were inconsistent. I've fixed this up and it works now. Cheers, Cedd

mpizenberg commented:
> The ps1 script didn't work, as the parameters names were inconsistent. I've fixed this up and it works now. Cheers, Cedd

Sorry I changed one line and couldn't do it right 😅. Thanks for the fix!
